### PR TITLE
feat(label): added animation style property to apply it to circular scrolling animation of label widget

### LINKED ISSUE #3097: Support for start and repeat delay for circular scrolling labels
### Introduce the problem

LVGL should support these kinds of animations out of the box:
ezgif com-gif-maker

The gif above shows a label that is displayed at the starting position for X seconds (animation start delay), then it does the circular scrolling animation and when it comes back to the starting position it stays there for Y seconds (repeat delay) before starting to scroll again.

See LVGL forum post [here](https://forum.lvgl.io/t/how-to-customize-the-circular-scrolling-of-a-label/8011/14).

### Examples and cases
See the gif above.

### Suggested solution
I think there needs to be a way to set the `start_delay` and `repeat_delay` properties of the scrolling animation.

### PR DESCRIPTION
**Pull request by andresovela**
### Description of the feature or fix

Added an animation style property to be used as animation template for different use cases in different widgets.
This commit implements using this property to set the start and repeat delay of the circular scrolling animation of the label widget.

Closes #3097.

I think the new `ANIM` property should replace `ANIM_TIME` wherever that is used, but I think that's beyond the scope of this PR.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [x] Update the documentation

#### Note
I'm not sure what documentation needs updating.

### CODE REVIEW

File: scripts/style_api_gen.py [hunk 1]
```diff
@@ -344,6 +344,10 @@
  'style_type': 'num',   'var_type': 'lv_opa_t' ,  'default':'`LV_OPA_TRANSP`', 'inherited': 0, 'layout': 0, 'ext_draw': 0,
  'dsc': "The intensity of mixing of color filter."},
 
+ {'name': 'ANIM',
+ 'style_type': 'ptr',   'var_type': 'const void *',  'default':'`NULL`', 'inherited': 0, 'layout': 0, 'ext_draw': 0,
```
Comment (kisvegabor) [inline scripts/style_api_gen.py]:
```suggestion
 'style_type': 'ptr',   'var_type': 'const lv_anim_t *',  'default':'`NULL`', 'inherited': 0, 'layout': 0, 'ext_draw': 0,
```

**Review by kisvegabor — COMMENTED** [review]
Very clean implementation, thank you!

Please also write a few words about it in the [docs of the label widgets](https://github.com/lvgl/lvgl/blob/master/docs/widgets/core/label.md#parts-and-styles).

It'd be even better to add an [example](https://github.com/lvgl/lvgl/tree/master/examples/widgets/label) with your special delayed animation to show case this feature.

**Comment by andresovela** [comment]
Done. I only need help with the Python version of the example.

**Comment by kisvegabor** [comment]
@andresovela 
Thanks you! There same issues causing CI failure:
> /home/runner/work/lvgl/lvgl/examples/widgets/label/lv_example_label_5.c:8:6: error: no previous prototype for ‘lv_example_label_5’ [-Werror=missing-prototypes]

`void lv_example_label_5(void);` should be added [here](https://github.com/lvgl/lvgl/blob/master/examples/widgets/lv_example_widgets.h#L84)


> Code format

Please run the code formatter: run `./code-formatter.py` from the `scrips` folder.





@uraich Can you help us out with the Python example? :slightly_smiling_face:

**Comment by andresovela** [comment]
Done

**Comment by kisvegabor** [comment]
And all test passed, merging.

Thank you!

**Comment by andresovela** [comment]
Just a reminder that the Python example is still missing

**Comment by uraich** [comment]
I just did it. You can find the code here:
https://github.com/uraich/lv_mpy_examples_v8/blob/main/widgets/label/lv_example_label_5.py
I will first have a look at label example 4 before creating a pull request

**Comment by kisvegabor** [comment]
Thanks @uraich! Let me know i you have any questions about the 4th example. It's a tricky one. :slightly_smiling_face:

**Pull request merged**